rustls/src/server/server_conn.rs

```diff
@@ -196,6 +226,20 @@ impl<'a> ClientHello<'a> {
     pub fn cipher_suites(&self) -> &[CipherSuite] {
         self.cipher_suites
     }
+
+    /// Get the server certificate types offered in the ClientHello.
+    ///
+    /// Returns `None` if the client did not include a certificate type extension.
+    pub fn server_cert_types(&self) -> Option<&'a [CertificateType]> {
```

Comment:
nit: I think it might be more natural to name this `server_cert_type()` since it's a singular `CertificateType` optionally returned? WDYT?

(If you agree, same feedback applies to `client_cert_types()`).

Reply:
It's not necessarily the case that only a singular `CertificateType` is returned. When a client supports both x509 and rpk's a both `CertificateType`s are sent in the extension. The 'equivalent' function for handling server extensions, `HasServerExtensions::client_cert_type`, does always return a singular `CertificateType`.

Reply:
Ahh! Ok, that makes sense. Thanks for explaining.